server/channels/app/notification.go

```diff
@@ -914,52 +918,65 @@ func (a *App) FilterUsersByVisible(c request.CTX, viewer *model.User, otherUsers
 	return result, nil
 }
 
-func (a *App) filterOutOfChannelMentions(c request.CTX, sender *model.User, post *model.Post, channel *model.Channel, potentialMentions []string) ([]*model.User, []*model.User, error) {
+func (a *App) filterOutOfChannelMentions(c request.CTX, sender *model.User, post *model.Post, channel *model.Channel, potentialMentions []string) ([]*model.User, []*model.User, []*model.User, error) {
 	if post.IsSystemMessage() {
-		return nil, nil, nil
+		return nil, nil, nil, nil
 	}
 
 	if channel.TeamId == "" || channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
-		return nil, nil, nil
+		return nil, nil, nil, nil
 	}
 
 	if len(potentialMentions) == 0 {
-		return nil, nil, nil
+		return nil, nil, nil, nil
 	}
 
-	users, err := a.Srv().Store().User().GetProfilesByUsernames(potentialMentions, &model.ViewUsersRestrictions{Teams: []string{channel.TeamId}})
+	mentionedUsersInTheTeam, err := a.Srv().Store().User().GetProfilesByUsernames(potentialMentions, &model.ViewUsersRestrictions{Teams: []string{channel.TeamId}})
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	// Filter out inactive users and bots
-	allUsers := model.UserSlice(users).FilterByActive(true)
-	allUsers = allUsers.FilterWithoutBots()
-	allUsers, appErr := a.FilterUsersByVisible(c, sender, allUsers)
+	teamUsers := model.UserSlice(mentionedUsersInTheTeam).FilterByActive(true)
+	teamUsers = teamUsers.FilterWithoutBots()
+	teamUsers, appErr := a.FilterUsersByVisible(c, sender, teamUsers)
+	if appErr != nil {
+		return nil, nil, nil, appErr
+	}
+
+	allMentionedUsers, err := a.Srv().Store().User().GetProfilesByUsernames(potentialMentions, nil)
+	if err != nil {
+		return nil, nil, nil, err
+	}
```

Comment:
We're calling `GetProfilesByUsernames` two times now instead of one. Is there a way call it only once, getting all users, and then filter by team manually to get only the team users?

Reply:
Actually we don't have associated team info in the user object. That's we can't filter with teamId.

Reply:
Ugh, you're absolutely right, that's a bummer. Let's go with it, given it's a fairly low-frequency call. Thanks for checking, @hossain-sazzad!